### ISSUE TITLE
Rules on wordpress-htaccess.conf seems not to be working

### ISSUE THREAD

**Issue by ambrosiora** [bitnami/bitnami-docker-wordpress#244]
Hi everyone. First of all, Thanks for provide this image.
__
Before start, I have to warn that I'm in learning process, please if there something I'm doing wrong, just tell me. I really need to ask for help, because the next step I will do is configure some plugin htaccess rules on wordpress-htaccess.conf
__

After do some researches about .htaccess on bitnami images, I found out that for security and performance reasons, the configurations are placed in `/opt/bitnami/wordpress/wordpress-htaccess.conf`.
Is that right?

Well, I'm trying to **check if the conditions witted down in this file is taking effect**, but It seems to not.
Or I'm doing something really wrong. (sorry)

On my host machine, my docker-compose.yml have this environment variable for the wordpress container:
`- WORDPRESS_HTACCESS_PERSISTENCE_ENABLED=yes`
and after mount a volume on:
`- ./persisted_files:/bitnami/wordpress`
The `wordpress-htaccess.conf` was able to edit.

So, I'm trying to `deny from all` the document root (`/opt/bitnami/wordpress`)
only for testing if when I access my website will appear a forbidden message or something.
If this happens, I will know that this rules are loaded and working well.

So I edited the `wordpress-htaccess.conf` and on the last lines added this above the comment "#tests purposes":
Check my entire file is:

```
<Directory "/opt/bitnami/wordpress/wp-content/plugins/akismet">
  # Only allow direct access to specific Web-available files.

# Apache 2.2
<IfModule !mod_authz_core.c>
        Order Deny,Allow
        Deny from all
</IfModule>

# Apache 2.4
<IfModule mod_authz_core.c>
        Require all denied
</IfModule>

# Akismet CSS and JS
<FilesMatch "^(form\.js|akismet\.js|akismet\.css)$">
        <IfModule !mod_authz_core.c>
                Allow from all
        </IfModule>

        <IfModule mod_authz_core.c>
                Require all granted
        </IfModule>
</FilesMatch>

# Akismet images
<FilesMatch "^logo-full-2x\.png$">
        <IfModule !mod_authz_core.c>
                Allow from all
        </IfModule>

        <IfModule mod_authz_core.c>
                Require all granted
        </IfModule>
</FilesMatch>
  </Directory>

#tests purposes
<Directory "/opt/bitnami/wordpress">
        Order Deny,Allow
        Deny from all
</Directory>
```

after this changes, do `docker-compose down && docker-compose up -d` to really force destroy and recreate container.
Just remembering that I have a volume on `/bitnami/wordpress`, so it's ok to run docker-compose down :)

But, it seems that `deny from all` was not been applied. My page still load normal.
**Can anyone help me to "debug" this?**

----
NOTE:
I Know If I change another environment variable `WORDPRESS_HTACCESS_OVERRIDE_NONE=no`
then, the default `.htaccess` file will be created instead of `wordpress-htaccess.conf` and `deny from all` works fine. I've tested it. But, I don't want to use it this way.

All I want to do is check if wordpress-htaccess.conf is being loaded to move to next step I've planned.
(Copy htaccess rules created from W3 Total Cache and paste inside a new `<Directory ...>` on the wordpress-htaccess.conf

Thanks!

**Comment by andresbono** [comment]
Hi @ambrosiora, thanks for opening this issue. You have discovered a bug in the configuration logic of the container!

In order to workaround it, please mount an additional volume like this:

```
- '/path/to/wordpress-htaccess.conf:/opt/bitnami/apache/conf/vhosts/htaccess/wordpress-htaccess.conf'
```
Can you confirm if it works for you? In the meantime, we will try to find a solution for the issue.

For you to understand what is happening, the file that is actually included by Apache is `/opt/bitnami/apache/conf/vhosts/htaccess/wordpress-htaccess.conf`. This file is created by copying the file from `/opt/bitnami/wordpress/wordpress-htaccess.conf` during the initialization of the container. The problem is that it is being copied **before** the persisted `wordperss-htaccess.conf` is restored, so your changes in the persisted file are not being applied.

**Comment by ambrosiora** [comment]
Wow @andresbono, thanks for your support!

After follow your instructions, to workaround it, **It worked well**! so far so good.

I repeated the same `deny from all` test and it works now.
And Thank you for explanation about initialization process! 

---
Just to help the others, here's my two cents:
After mount the directory: `/opt/bitnami/apache/conf/vhosts/htaccess/` on host, and edit the `wordpress-htaccess.conf` file, you need to restart apache, which I achieved this way:

On host:
`docker exec -it <CONTAINER_NAME> /bin/bash`

on container:
`apachectl restart`

**Comment by rXc3NtR1c** [comment]
I'll chime in here too, I'm using the file to add php directives for file size so I don't have to rebuild the container with an updated php.ini.

I'm going to state the (now) obvious - you need to edit both files with the same changes. The file still gets copied, so if you restart you may run into issues if you only change the apache one.

**Comment by ambrosiora** [comment]
Hi, thanks!

Just saying, I moved to (Nginx) https://github.com/bitnami/bitnami-docker-wordpress-nginx
And, my experiences was very satisfying! Due to Nginx and its non-blocking I/O behavior, the performance was increased! 🚀

**Comment by marcosbc** [comment]
Hi all, we have just released tag [`5.7.1-debian-10-r21`](https://github.com/bitnami/bitnami-docker-wordpress#571-debian-10-r21) of the container images which implements a lot of new features to this container image.

You may want to look into these [environment variables](https://github.com/bitnami/bitnami-docker-wordpress#user-and-site-configuration):

* `WORDPRESS_HTACCESS_OVERRIDE_NONE`: Set the Apache AllowOverride variable to None. All the default directives will be loaded from `opt/bitnami/wordpress/wordpress-htaccess.conf`. Default: `yes`
* `WORDPRESS_ENABLE_HTACCESS_PERSISTENCE`: Persist the custom changes of the htaccess. It depends on the value of `WORDPRESS_HTACCESS_OVERRIDE_NONE`, when yes it will persist `/opt/bitnami/wordpress/wordpress-htaccess.conf` if no it will persist `/opt/bitnami/wordpress/.htaccess`. Default: `no`

FYI @ambrosiora, we also added [most of the improvements](https://github.com/bitnami/bitnami-docker-wordpress-nginx#571-debian-10-r22) to the wordpress-nginx image!
